### DISCUSSION TITLE
How to disable a certain rule sid of Suritaca

### DISCUSSION THREAD

**Discussion by ldceven** [Security-Onion-Solutions/securityonion#2151] [Unsupported Versions] [answered]
Hello, my environment is distributed deployment,The node information is as follows:
[root@dc-sec-so-web-01 minions]# sudo salt \* test.ping
dc-sec-so-sensor-01_sensor:
    True
dc-sec-so-search-01_searchnode:
    True
dc-sec-so-web-01_managersearch:
    True
dc-sec-so-search-02_searchnode:
    True

Try to disable sid on the management node according to the official website document, and change the configuration "/opt/so/saltstack/local/pillar/minions/<minionid>.sls"

idstools:
  config:
    ruleset:'ETOPEN'
    oinkcode:''
    urls:
  sids:
    enabled:
    disabled: 2010937
    modify:
kratos:
  kratoskey:'SAuEJMKmPb8x6W3oYGiC'

Then, from the manager run "sudo salt $SENSORNAME_$ROLE state.apply idstools "to update the config error.

[root@dc-sec-so-web-01 minions]# sudo salt dc-sec-so-sensor-01_sensor state.apply idstools
dc-sec-so-sensor-01_sensor:
----------
          ID: idstools_state_not_allowed
    Function: test.fail_without_changes
      Result: False
     Comment: Failure!
     Started: 06:12:52.224436
    Duration: 3.703 ms
     Changes:

Summary for dc-sec-so-sensor-01_sensor
------------
Succeeded: 0
Failed: 1
------------
Total states run: 1
Total run time: 3.703 ms
ERROR: Minions returned with non-zero exit code

Thanks for your help！

**Comment by bioteseo** [comment]
Try
sudo so-status
on your sensor. You'll see the cause of failed above.

For now, I don't know in future but I hope this will be change, in a distributed deployment, you can disable/modify signatures only on your  manager (where idstools container run).

If you use search nodes, I think you have to use a manager node and not a managersearch node

**Reply by ldceven** [reply]
There seems to be no problem with the services of the sensor node
<img width="1399" alt="sensor-01" src="https://user-images.githubusercontent.com/67267374/100721241-0bef3a00-33fa-11eb-86e0-119765d46aaf.png">

The managersearch node also has idstools and is running normally
<img width="1390" alt="so-web-01" src="https://user-images.githubusercontent.com/67267374/100721276-16113880-33fa-11eb-95ff-b3410dea9022.png">

Must it be a manager node?

And I tried to disable sid by configuring disablesid.conf, but after a few minutes, the disablesid.conf configuration will be restored, I don’t know why

**Comment by dougburks** [comment]
Have you tried running `sudo salt $SENSORNAME_$ROLE state.apply idstools` replacing `$SENSORNAME_$ROLE` with the name and role of the **manager**?

**Reply by ldceven** [reply]
_Hello, I found that the roles name of the node is null_

[root@dc-sec-so-web-01 salt]# salt '*' grains.item roles
dc-sec-so-sensor-01_sensor:
    ----------
    roles:
dc-sec-so-search-01_searchnode:
    ----------
    roles:
dc-sec-so-web-01_managersearch:
    ----------
    roles:
dc-sec-so-search-02_searchnode:
    ----------
    roles:
[root@dc-sec-so-web-01 salt]#

**Reply by dougburks** [reply]
Have you tried running `sudo salt dc-sec-so-web-01_managersearch state.apply idstools`?

**Reply by ldceven** [reply]
> Have you tried running `sudo salt dc-sec-so-web-01_managersearch state.apply idstools`?

I tried running, but an error occurred

[root@dc-sec-so-web-01 minions]# sudo salt dc-sec-so-web-01_managersearch state.apply idstools
dc-sec-so-web-01_managersearch:
----------
          ID: idstoolsdir
    Function: file.directory
        Name: /opt/so/conf/idstools/etc
      Result: True
     Comment: The directory /opt/so/conf/idstools/etc is in the correct state
     Started: 02:03:21.837068
    Duration: 18.271 ms
     Changes:   
----------
          ID: idstoolslogdir
    Function: file.directory
        Name: /opt/so/log/idstools
      Result: True
     Comment: The directory /opt/so/log/idstools is in the correct state
     Started: 02:03:21.855671
    Duration: 3.933 ms
     Changes:   
----------
          ID: idstoolsetcsync
    Function: file.recurse
        Name: /opt/so/conf/idstools/etc
      Result: False
     Comment: #### /opt/so/conf/idstools/etc/disable.conf ####
              Unable to manage file: Jinja error: 'int' object is not iterable
              Traceback (most recent call last):
                File "/usr/lib/python3.6/site-packages/salt/utils/templates.py", line 498, in render_jinja_tmpl
                  output = template.render(**decoded_context)
                File "/usr/lib/python3.6/site-packages/jinja2/environment.py", line 1090, in render
                  self.environment.handle_exception()
                File "/usr/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
                  reraise(*rewrite_traceback_stack(source=source))
                File "/usr/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
                  raise value.with_traceback(tb)
                File "<template>", line 11, in top-level template code
              TypeError: 'int' object is not iterable
              
              ; line 11
              
              ---
              [...]
              # 2019401
              
              # Example of disabling a rule by regular expression.
              # - All regular expression matches are case insensitive.
              # re:hearbleed
              # re:MS(0[7-9]|10)-\d+    <======================
              {%- if disabled_sids != None %}
              {%- for sid in disabled_sids %}
              {{ sid }}
              {%- endfor %}
              {%- endif %}
              [...]
              ---
     Started: 02:03:21.859882
    Duration: 168.079 ms
     Changes:   
----------
          ID: so-ruleupdatecron
    Function: cron.present
        Name: /usr/sbin/so-rule-update > /opt/so/log/idstools/download.log 2>&1
      Result: True
     Comment: Cron /usr/sbin/so-rule-update > /opt/so/log/idstools/download.log 2>&1 already present
     Started: 02:03:22.031733
    Duration: 16.319 ms
     Changes:   
----------
          ID: rulesdir
    Function: file.directory
        Name: /opt/so/rules/nids
      Result: True
     Comment: The directory /opt/so/rules/nids is in the correct state
     Started: 02:03:22.048373
    Duration: 2.72 ms
     Changes:   
----------
          ID: synclocalnidsrules
    Function: file.recurse
        Name: /opt/so/rules/nids/
      Result: True
     Comment: The directory /opt/so/rules/nids/ is in the correct state
     Started: 02:03:22.051266
    Duration: 99.725 ms
     Changes:   
----------
          ID: so-idstools
    Function: docker_container.running
      Result: False
     Comment: One or more requisite failed: idstools.idstoolsetcsync
     Started: 02:03:22.242684
    Duration: 0.008 ms
     Changes:   
----------
          ID: append_so-idstools_so-status.conf
    Function: file.append
        Name: /opt/so/conf/so-status/so-status.conf
      Result: True
     Comment: File /opt/so/conf/so-status/so-status.conf is in correct state
     Started: 02:03:22.242789
    Duration: 7.22 ms
     Changes:   

Summary for dc-sec-so-web-01_managersearch
------------
Succeeded: 6
Failed:    2
------------
Total states run:     8
Total run time: 316.275 ms
ERROR: Minions returned with non-zero exit code
[root@dc-sec-so-web-01 minions]#

**Reply by ldceven** [reply]
It feels like a file configuration format problem. May I ask the /opt/so/saltstack/local/pillar/minions/dc-sec-so-web-01_managersearch.sls file. What is the correct configuration format? I tried the following two methods to no avail

idstools:
   sids:
     enabled:
     disabled: 2010937
     modify:
or
idstools:
   sids:
     enabled:
     disabled:
            -2010937
     modify:

**Reply by greatapoc** [reply]
Put a space between the hyphen and the sid and ensure you're using the correct spacing with spaces instead of tabs. `- 2010937`

**Reply by ldceven** [reply]
Is it possible to disable sid in other ways? I am really unfamiliar with the use and configuration of saltstack. For example, by configuring /opt/so/conf/idstools/etc/disable.conf? But restart the so-idstools service, the disable.conf configuration will be lost, restore the original configuration

**Reply by scottpinkston** [reply]
seeing the same error here: trying with a single space between : and - and a single space between - and 2101411
Also same error if I move it down a line.

  disabled: - 2101411
  and
  disabled:
 - 2101411

**Reply by gcsrellis** [reply]
On the manager node in /opt/so/saltstack/local/pillar/global.sls  you pasted?
idstools:
&nbsp;  sids:
&nbsp;  &nbsp;  disabled:
&nbsp; &nbsp; &nbsp; - 2101411

**Reply by scottpinkston** [reply]
No - /opt/so/saltstack/local/pillar/minions/wazuh_standalone.sls

idstools:
  sids:
    enabled:
    disabled:
     - 2101411
    modify:

**Reply by gcsrellis** [reply]
Are you running a distributed deployment?  The original poster mentioned he was and that's where it needs to be.  The global pillar on the managersearch node, not the minion pillar as he stated on the first post.

**Comment by ldceven** [comment, marked as answer]
I found the correct configuration, please close this question。

![WX20201208-010006](https://user-images.githubusercontent.com/67267374/101380695-cdbdb300-38f0-11eb-99ed-679442ae48e4.png)

![WX20201208-010026](https://user-images.githubusercontent.com/67267374/101380730-d615ee00-38f0-11eb-8d60-e60c932d0143.png)

![WX20201208-010349](https://user-images.githubusercontent.com/67267374/101381081-458bdd80-38f1-11eb-8517-2b4301acb91c.png)

**Comment by rwwiv** [comment]
@ldceven 
We're currently in the process of writing a tool to make disabling SIDs easier. If you'd like to test it out early, copy [this link's content](https://raw.githubusercontent.com/Security-Onion-Solutions/securityonion/feature/so-rules/salt/common/tools/sbin/so-rules) into `/usr/sbin/so-rules` on the manager (or alternatively, copy the script from the `feature/so-rules` branch of this repo - found under `securityonion/salt/common/tools/sbin`), then copy the following script somewhere onto the same system and run it as root:
``` sh
#!/bin/bash

. /usr/sbin/so-common

# Surround double curly braces in any minion pillar in single quotes so the pillars can be parsed as regular yaml

readarray -t minion_pillars <<< "$(find /opt/so/saltstack/local/pillar/minions -type f -name '*.sls')"
for pillar in "${minion_pillars[@]}"; do
  sed -i -r "s/ (\{\{.*}})$/ '\1'/g" "$pillar" 
done
```
The tool has built in help with `-h` to assist you with syntax.

**Reply by greatapoc** [reply]
I've just given it a quick test and it seems to work fine. Adding a sid to disable was simple, haven't tried any regex. 

Does it matter that it is wrapping the sid in apostrophes in the minion file? Here's an excerpt of mine (script added last rule)

``` 
- 2028781  
- 2002192  
- '2029575'  
enabled: null  
modify: null  
```

**Comment by TOoSmOotH** [comment]
Wrapping it in ticks makes it a string. Because you are using ints in the first 2 its ok. Where you would run into problems is if you were doing something like re:1234.* by hand to the sls  would cause errors. The tool will add this as 're:1234.*'

**Reply by greatapoc** [reply]
Makes sense, thanks for that. I guess eventually this will all be abstracted away and there will be little reason to look at it as long as it works.

Totally understand this is just the first step but will it also extend to src/dst suppressions?

**Reply by isavickas** [reply]
Would adding comments after the sid number cause any issues?  

i.e.:

- 2028781   # Disabled on date by user for reason
